src/components/ProjectDetails/ProjectDetails.js

```diff
@@ -152,15 +158,15 @@ const ProjectDetails = ( {
             <Button
               level="neutral"
               text={t( "JOIN" )}
-              onPress={joinProject}
+              onPress={() => setOpenSheet( JOIN )}
               loading={loadingProjectMembership}
             />
           )
           : (
             <Button
               level="neutral"
               text={t( "LEAVE" )}
-              onPress={leaveProject}
+              onPress={() => setOpenSheet( LEAVE )}
               loading={loadingProjectMembership}
```

Comment:
Same here, I think we should disable this button while the leave request is in flight.

Reply:
Done